File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/SetDocumentOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestoreHelper;
import java.util.Map;
import org.json.JSONException;

public class SetDocumentOptions {

    private String reference;
    private Map<String, Object> data;
    private boolean merge;
    @Nullable
    private String databaseId;

    public SetDocumentOptions(String reference, JSObject data, boolean merge, @Nullable String databaseId) throws JSONException {
        this.reference = reference;
        this.data = FirebaseFirestoreHelper.createHashMapFromJSONObject(data);
        this.merge = merge;
        this.databaseId = databaseId;
    }

    public String getReference() {
        return reference;
    }

    public Map<String, Object> getData() {
        return data;
    }

    public boolean getMerge() {
        return merge;
    }

    @Nullable
    public String getDatabaseId() {
        return databaseId;
    }
}

File: packages/firestore/ios/Plugin/Classes/Options/SetDocumentOptions.swift
import Foundation
import Capacitor

@objc public class SetDocumentOptions: NSObject {
    private var reference: String
    private var data: [String: Any]
    private var merge: Bool
    private var databaseId: String?

    init(reference: String, data: JSObject, merge: Bool, databaseId: String? = nil) {
        self.reference = reference
        self.data = FirebaseFirestoreHelper.createHashMapFromJSObject(data)
        self.merge = merge
        self.databaseId = databaseId
    }

    func getReference() -> String {
        return reference
    }

    func getData() -> [String: Any] {
        return data
    }

    func getMerge() -> Bool {
        return merge
    }

    func getDatabaseId() -> String? {
        return databaseId
    }
}
